_projects/2024/100431421/100431421.Rmd

```diff
@@ -0,0 +1,666 @@
+
+Here I tried to apply Iñaki's advice to improve the plot, specifically I take into account these suggestion:
+
+>
+1. Avoid categories and make the "% of immigrants" continuous since we have this data. 
+
+>
+2. Flip the color scale (blue for low values and yellow for high values)
+
+
+**PROBLEM**: If we do the plot continuous the map would look blue because the majority of countries have less than 25% of immigration, as we can see:
+
+```{r}
+linear_scale_colors <- scale_fill_gradientn(
+  colors = c("#286b80", "#388f8d", "#72b56e", "#aac25c", "#eef100"), 
+  na.value = "gray80",  
+  name = "% of immigrants",  
+  limits = c(0, 100),
+  breaks = c(0, 25, 50, 75, 100),
+  labels = c("0%", "25%", "50%", "75%", "100%")  
+)
+
+map_cont <- ggplot() +
+  geom_sf(data = datos_mapa, 
+          aes(fill = Percent_Migration), 
+          color = "white", linewidth = 0.2)+
+  theme_void() +
+  linear_scale_colors+
+  theme(
+    legend.position = c(0.2, 0.4),
+    legend.title = element_text(size = 10, face = "bold"),
+    legend.key.size = unit(0.15, "cm"),
+    legend.key.height = unit(0.25, "cm"),
+    legend.key.width = unit(0.25, "cm"),
+    legend.text = element_text(size = 8),
+    aspect.ratio = 0.6
+  ) +
+  coord_sf()
+
+map_cont
+```
+
+In this map we can not see differences between countries, only in the Arabian zone where immigration is notably higher. A better option would be a logarithmic scale since it would better highlight variations across regions with lower immigration rates. So, we could try:
+
+```{r}
+log_scale_colors <- scale_fill_gradientn(
+  colors = c("#286b80", "#388f8d", "#72b56e", "#aac25c", "#eef100"), 
+  trans = "log10",  # logarithmic scale
+  na.value = "gray80",  
+  limits = c(0.01, 100),
+  name = "% of immigrants",  
+  breaks = c(0.01, 1, 5, 25, 100),  #since it's logarithmic we can not star from 0
+  labels = c("0.01%", "1%", "5%", "25%", "100%")  
+)
+
+# logarithmic MAP
+map_log <- ggplot() +
+  geom_sf(data = datos_mapa, 
+          aes(fill = Percent_Migration), 
+          color = "white", linewidth = 0.2) +
+  log_scale_colors +
+  theme_void() +
+  theme(
+    legend.position = c(0.2, 0.4),
+    legend.title = element_text(size = 8, face = "bold"),
+    legend.key.size = unit(0.15, "cm"),
+    legend.key.height = unit(0.25, "cm"),
+    legend.key.width = unit(0.25, "cm"),
+    legend.text = element_text(size = 6),
+    aspect.ratio = 0.6
+  ) +
+  coord_sf()
+
+map_log
+```
+
+**Before continuing with the plot (adding the migratory flows, barplots, sources, titles...), I have some questions to know how to proceed with the map scale**
+
+# QUESTIONS
+
+First, I think it's better -as you said- that the blue and dark colors are related to lower values and the higher values are represented with more lighter and yellowish colors. But what do you think about the logarithmic scale? I think that thanks to it we can highlight better the differences between countries than with the continuous scale.
```

Comment:
And please try to avoid things like "as you said", because this post is not for me, it's also for you, it's for everyone! So imagine some random person arrives to your project and reads this... :)